webui/src/components/common/categorySelect.vue

```diff
@@ -0,0 +1,116 @@
+<script setup>
+import { ref, watch, computed } from 'vue'
+import TreeSelect from 'primevue/treeselect'
+import { useCategoryTree } from '@/composables/useCategoryTree'
+
+const props = defineProps({
+    modelValue: { type: Number, default: null }, // parentId
+    type: { type: String, default: 'expense' } // expense | income
+})
+
+const { IncomeTreeData, ExpenseTreeData } = useCategoryTree()
+const emit = defineEmits(['update:modelValue'])
+
+/* ---------------------------
+   Helpers
+---------------------------- */
+// Pick correct tree
+const rawTreeData = computed(() => {
+    return props.type === 'expense' ? ExpenseTreeData.value : IncomeTreeData.value
+})
+
+// Convert raw nodes → TreeSelect format
+const convertTree = (nodes, parentKey = '0', parentPath = '') => {
+    if (!nodes || !Array.isArray(nodes)) return []
+
+    return nodes.map((node, index) => {
+        const currentKey = `${parentKey}-${index}`
+        const path = parentPath ? `${parentPath} / ${node.data.name}` : node.data.name
+
+        const converted = {
+            key: String(node.data.id),
+            label: node.data.name,
+            data: { ...node.data, path }
+        }
+
+        if (node.children?.length) {
+            converted.children = convertTree(node.children, currentKey, path)
+        }
+
+        return converted
+    })
+}
+
+const categoryTreeData = computed(() => {
+    return [
+        { key: 'root', label: 'Root Category', checked: true },
```

Comment:
for the backend to work root `key` needs to be 0,
e.g. https://github.com/andresbott/etna-finance/blob/b51fd0e4154ae3b39ac579cab3ec0c387983da30/webui/src/views/categories/dialogs/CategoryDialog.vue#L79-L89